### ISSUE TITLE
training on single GPU is very slow because GPU is rarely used

### ISSUE THREAD

**Issue by gitunit** [maudzung/TTNet-Real-time-Analysis-System-for-Table-Tennis-Pytorch#5]
not sure why training is so slow, I'm using a single GPU but nvidia-smi shows me that the GPU is rarely used.
could it be that "cv2.resize" is the bottleneck? I'm new to pytorch but i assume the dataloader is resizing the data for each batch before training, is that correct?

**Comment by maudzung** [comment]
hi @gitunit 
Did you change any lines in my code? What's your type of GPU?

**Comment by gitunit** [comment]
i haven't changed anything except the image path, im on Quadro RTX 3000. im running phase 1 training script. it doesnt seem to be GPU type related since it's rarely used, does it?

**Comment by maudzung** [comment]
Can you give me more information: Batch_size, num_workers...
Did you run on a single GPU? If so, please refer to the instruction [here](https://github.com/maudzung/TTNet-Real-time-Analysis-System-for-Table-Tennis-Pytorch/blob/master/README.md#2311-single-machine-single-gpu)

**Comment by gitunit** [comment]
yes, single GPU. i used same configs as in "./train_1st_phase.sh".
i tried running "python main.py --gpu_idx 0" but did run out of memory.

**Comment by maudzung** [comment]
I guess the problem may stem from the loading data phase. What is your disk type, is it HDD or SSD? Because an input tensor of the TTNet is composed of _9 sequential images_, it may be slow if you load images from HDD. 
In the code of the dataset loader, I tried to make the dataloader faster by using TurboJPEG instead of cv2. 
I ran the experiments on Ubuntu 18.04, GTX 1080Ti, SSD disk... , each epoch lasts around 8-10 minutes (using smooth labeling events).
P/S: You also can try to change the num_workers, the bigger number of workers may not make the training phase faster. Read more at [here](https://discuss.pytorch.org/t/guidelines-for-assigning-num-workers-to-dataloader/813/28)
